### PR TITLE
Generate CNP file

### PR DESCRIPTION
**Pull request by david-gimelle**
### Change description ###

Add a new CNP file when a project is generated

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: Jenkinsfile_template [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+@Library("Infrastructure")
+
+def type = "java"
+def product = "rpe"
+def component = "demo"
+
+
+withPipeline(type, product, component) {
+    installCharts()
```
Comment (timja) [inline Jenkinsfile_template:20]: this no longer does anything
```suggestion
```

File: Jenkinsfile_template [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!groovy
+properties(
+    [
+        [
+            $class       : 'GithubProjectProperty',
+            projectUrlStr: 'https://github.com/hmcts/spring-boot-template'
+        ],
+        pipelineTriggers([[$class: 'GitHubPushTrigger']])
+    ]
+)
```
Comment (timja) [inline Jenkinsfile_template:10]: also not needed
```suggestion
```

File: bin/init.sh [hunk 3]
```diff
@@ -13,6 +13,7 @@ read -p "Port number for new app: " port
 read -p "Replace \`demo\` package name with: " package
 read -p "Repo product: (It's first part of the git repo name. Often a team name) " product_name
 read -p "Repo component: (It's second part of git repo name. Application name) " component_name
+read -p "Slack Channel for build notification " slack_channel
```
Comment (timja) [inline bin/init.sh:16]: this should really be handled in the project config, shouldn't be needed in build config any more
https://github.com/hmcts/cnp-jenkins-config/blob/master/team-config.yml#L6

File: Jenkinsfile_template [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    onMaster {
+        enableSlackNotifications('#rpe-spring-demo-builds')
+    }
+    onDemo {
+        enableSlackNotifications('#rpe-spring-demo-builds')
+    }
+    onPR {
+        enableSlackNotifications('#rpe-spring-demo-builds')
+    }
```
Comment (timja) [inline Jenkinsfile_template:31]: I would recommend against this as mentioned in https://github.com/hmcts/spring-boot-template/pull/187/files#r427355521
```suggestion
```

File: Jenkinsfile_template (hunk 2 above)
Comment (david-gimelle) [reply Jenkinsfile_template]: done

File: Jenkinsfile_template (hunk 1 above)
Comment (david-gimelle) [reply Jenkinsfile_template]: done

File: Jenkinsfile_template (hunk 4 above)
Comment (david-gimelle) [reply Jenkinsfile_template]: done

File: bin/init.sh (hunk 3 above)
Comment (david-gimelle) [reply bin/init.sh]: ok

**Pull request merged**